c10/cuda/CUDACachingAllocator.cpp

```diff
@@ -790,6 +791,15 @@ cudaError_t cudaMallocMaybeCapturing(void** p, size_t size) {
 
 static std::string reportProcessMemoryInfo(int device) {
 #ifdef PYTORCH_C10_DRIVER_API_SUPPORTED
+  void* handle_1 = dlopen("libnvidia-ml.so.1", RTLD_LAZY);
+  try {
+    TORCH_INTERNAL_ASSERT(handle_1);
+  } catch (...) {
+    nvml_is_available = false;
+  }
+  if (!c10::cuda::nvml_is_available) {
```

Comment:
Would we want to indicate that NVML isn't available in this case e.g., via a warning? I can't tell if the user would be able to see anything in this case.

Reply:
I think the context here is that this function will return an empty string if nvml is not available and a string with actual useful information if it is available. This is then used in the out of memory reporting function here: https://github.com/pytorch/pytorch/blob/main/c10/cuda/CUDACachingAllocator.cpp#L1054, https://github.com/pytorch/pytorch/blob/main/c10/cuda/CUDACachingAllocator.cpp#L1126. But from the user's perspective, without nvml they still get a well-formed out of memory error message, just missing the extra information that nvml would report. Do we want to then add some kind of message in this case like "For more detailed out of memory reporting, please install nvml if possible" or something, or would this not matter to most users?